Plando: Allow >100 Gold Skulltula Token Requirements

Cuphat opened the pull request:
This allows you to plando Bridge/Ganon Boss Key/LACS token requirements up to 999 while still only allowing those to be set up to 100 in the GUI. If there aren't enough tokens available in a world to meet the requirements set, you will get a specific error message.

Cuphat committed:
Prioritize version-specific GUI parameters in SettingsToJson

This makes sure that if 'web:key' or 'electron:key' are specified in the 'gui_params' of a setting, the value specified is always used in the appropriate case over the one specified in 'key'.

Cuphat committed:
Plando: Allow >100 Gold Skulltula Token Requirements

This allows you to plando Bridge/Ganon Boss Key/LACS token requirements up to 999 while still only allowing those to be set up to 100 in the GUI. If there aren't enough tokens available in a world to meet the requirements set, you will get a specific error message.

r0bd0g commented:
I suggest the requirement for the skulltula icon to change colour should be when you have tokens equal to the number in the pool?

Cuphat committed:
Update Gold Skulltula Tokens highlighting in Pause Menu

Instead of turning the Gold Skulltula count in the Quest Status screen red at 100 tokens, it will now turn red at the maximum available tokens in the world.

Cuphat commented:
> I suggest the requirement for the skulltula icon to change colour should be when you have tokens equal to the number in the pool?

Done. 
https://cdn.discordapp.com/attachments/512048482677424138/932754379688271912/ootr_120_available_100_count.png
https://cdn.discordapp.com/attachments/512048482677424138/932754379902177370/ootr_120_available_120_count.png

cjohnson57 commented:
This seems good, just 2 minor requests:

1) If no `item_pool` is provided in the plando, could the pool automatically be adjusted to the value it needs to be, assuming it can be? 

2) It'd be good to add a couple unit tests for this, at least 1 that should error and 1 that shouldn't
